release/0.2.x candidate (#1126)
GEOSX/GEOSX: RELEASE_NOTES.md

Version v0.2.0 -- Release date 2020-06-20
==========================================
* Extensive updates to Data Repository.
* Discretization:
  * Finite Element Interface
    * Kernel launching interface looping abstraction.
    * Element Formulations for 1st order:
      * 8 node hexahedron
      * 6 node wedge
      * 5 node pyramid
      * 4 node tetrahedron

>   * Finite Volume Stencils (TPFA, infrastructure support for MPFA, but currently 

Author: @klevzoff @francoishamon Please elaborate

>   * Finite Volume Stencils (TPFA, infrastructure support for MPFA, but currently 
>     not implemented.)

Reviewer: ```suggestion
  * Finite Volume Stencils for Classical (cell-centered) FVM formulation (TPFA, infrastructure support for MPFA, but currently 
    not implemented)
  * Inner Products for Hybrid FVM formulation (current support for TPFA and quasi-TPFA) 
```

* Physics Solvers
  * Solid Mechanics Explicit on GPU, Implicit Assembly on GPU
  * Single Phase Flow (Assembly on GPU)

>   * Compositional Multiphase Flow (Assembly on GPU)

Author: @klevzoff Please elaborate

>   * Single Phase Flow (Assembly on GPU)
>   * Compositional Multiphase Flow (Assembly on GPU)

Reviewer 2: ```suggestion
  * Single Phase Flow (Assembly on GPU)
    * Classical FVM and Hybrid FVM formulations
    * Porous matrix and DFM fracture flow
  * Compositional Multiphase Flow (Assembly on GPU)
    * Fully Implicit Isothermal Overall Composition Formulation
    * A library of fluid constitutive models:
      * Equation-of-State based hydrocarbon compositional
      * Three-phase extended Black-oil
      * Two-phase CO2-brine  
```

>   * Compositional Multiphase Flow (Assembly on GPU)

Reviewer: ```suggestion
  * Compositional Multiphase Flow (Assembly on GPU)
  * Multi-segmented Wells for Single Phase and Compositional Multiphase Flow (Assembly on GPU)
```

  * Surface Generation
    * Topology change (Legacy GEOS approach)

>     * Enriched Finite Element Method

Author: @CusiniM Please elaborate

  * Hydrofracture Solver (Legacy GEOS approach)

>   * Small Strain aligned Contact using Lagrange Multipliers

Author: @af1990 Please elaborate.

* Mesh Structure
  * Introduced concept of extrinsic mesh data
  * Fracture Elements to represent FV cells in fracture

> * Proppant modeling

Author: @yue-2018 Please elaborate

> * VTK output

Author: @AntoineMazuyer Please elaborate

> * Linear Algebra Interface layers for Hypre, Trilinos, Petsc

Author: @castelletto1 Please elaborate

Reviewer 3: ```suggestion
* Linear Algebra Interface layers for Hypre, Trilinos, Petsc
  * Common interface for supported linear algebra packages
  * Krylov solvers (CG, GMRES, BiCGSTAB)
  * Preconditioners (Algebraic Multigrid, incomplete factorizations)
  * Block matrix and vector support
  * Serial and parallel direct solvers
```

Version v0.2.0 -- Release date 2020-06-20
==========================================
* Extensive updates to Data Repository.
* Discretization:
  * Finite Element Interface
    * Kernel launching interface looping abstraction.
    * Element Formulations for 1st order:
      * 8 node hexahedron
      * 6 node wedge
      * 5 node pyramid
      * 4 node tetrahedron
  * Finite Volume Stencils (TPFA, infrastructure support for MPFA, but currently 
    not implemented.)
* Physics Solvers
  * Solid Mechanics Explicit on GPU, Implicit Assembly on GPU
  * Single Phase Flow (Assembly on GPU)
  * Compositional Multiphase Flow (Assembly on GPU)
  * Surface Generation
    * Topology change (Legacy GEOS approach)

>     * Enriched Finite Element Method

Reviewer 4: ```suggestion
* Embedded Discrete Fractures
  * Enriched Finite Element Method for the discretization of the mechanics
  * Piecewise constant displacement jump enrichment
```

Version v0.2.0 -- Release date 2020-06-20
==========================================
* Extensive updates to Data Repository
* Discretization
  * Finite element interface
    * Kernel launching interface looping abstraction
    * Element formulations for 1st order:
      * 8-node hexahedron
      * 6-node wedge
      * 5-node pyramid
      * 4-node tetrahedron

>   * Finite volume stencils for cell-centered FVM formulations (TPFA, infrastructure support for MPFA, but currently 
>     not implemented)
>   * Inner products for Hybrid FVM formulations (current support for TPFA and quasi-TPFA) 

Reviewer 2: ```suggestion
  * Cell-centered Finite Volume method with Two-point Flux Approximation (TPFA)
  * Hybrid Finite Volume method with TPFA and quasi-TPFA inner products
```

* Physics Solvers
  * Solid mechanics explicit on GPU, implicit assembly on GPU
  * Single-phase flow (assembly on GPU)
    * Classical FVM and Hybrid FVM formulations
    * Porous matrix and DFM fracture flow

>   * Compositional multiphase flow (assembly on GPU)
>   * Multi-segmented wells for single phase and compositional multiphase flow (assembly on GPU)
>     * Fully implicit isothermal overall composition formulation
>     * Fluid constitutive models:
>       * Equation-of-state hydrocarbon compositional
>       * Three-phase extended black-oil
>       * Two-phase CO2-brine  

Reviewer 2: ```suggestion
  * Compositional multiphase flow (assembly on GPU)
    * Fully implicit isothermal overall composition formulation
    * Fluid constitutive models:
      * Equation-of-state hydrocarbon compositional
      * Three-phase extended black-oil
      * Two-phase CO2-brine
  * Multi-segmented wells for single phase and compositional multiphase flow (assembly on GPU)
```

Version v0.2.0 -- Release date 2020-06-20
==========================================
* Extensive updates to Data Repository.
* Discretization:
  * Finite Element Interface
    * Kernel launching interface looping abstraction.

>     * Element Formulations for 1st order:

Reviewer 5: This is totally up to you, but the capitalization style in these notes is all over the map.  Perhaps we can just capitalize the first word in each phrase unless it is a proper object like `Data Repository`.  Otherwise, I would just write `Finite volume stencils for classical FVM formulation`, `Element formulations for 1st order`, etc.  This obviously me being OCD.  We can leave as is if you prefer.

Author: I think I took care of this now.

      * 8 node hexahedron
      * 6 node wedge
      * 5 node pyramid
      * 4 node tetrahedron

>   * Finite volume stencils for classical (cell-centered) FVM formulation (TPFA, infrastructure support for MPFA, but currently 
>     not implemented)
>   * Inner products for Hybrid FVM formulation (current support for TPFA and quasi-TPFA) 

Reviewer 2: I think for release notes specifically there's no point in listing things that aren't yet implemented, so I would just keep it brief
```suggestion
  * Cell-centered Finite Volume method with Two-point Flux Approximation (TPFA)
  * Hybrid Finite Volume method with TPFA and quasi-TPFA inner products
```

Author: @klevzoff can you re-suggest this? It is out of date and I can't commit it.

Reviewer 2: Sure, done

* Physics Solvers
  * Solid mechanics explicit on GPU, implicit assembly on GPU
  * Single phase flow (assembly on GPU)
    * Classical FVM and Hybrid FVM formulations
    * Porous matrix and DFM fracture flow

>   * Compositional multiphase flow (assembly on GPU)
>   * Multi-segmented wells for single phase and compositional multiphase flow (assembly on GPU)
>     * Fully implicit isothermal overall composition formulation
>     * A library of fluid constitutive models:
>       * Equation-of-state based hydrocarbon compositional
>       * Three-phase extended black-oil
>       * Two-phase CO2-brine  

Reviewer 2: The order of bullet points got messed up a bit
```suggestion
  * Compositional multiphase flow (assembly on GPU)
    * Fully implicit isothermal overall composition formulation
    * A library of fluid constitutive models:
      * Equation-of-state based hydrocarbon compositional
      * Three-phase extended black-oil
      * Two-phase CO2-brine
  * Multi-segmented wells for single phase and compositional multiphase flow (assembly on GPU)
```

Author: @klevzoff can you re-suggest this? It is out of date and I can't commit it.

  * Surface Generation
    * Topology change (Legacy GEOS approach)
* Embedded Discrete Fractures
  * Enriched Finite Element Method for the discretization of the mechanics
  * Piecewise constant displacement jump enrichment
  * Hydrofracture Solver (Legacy GEOS approach)
  * Small Strain aligned Contact using Lagrange Multipliers
    * Discrete Fracture Model using a low order stabilized mixed finite element
  * Proppant Transport Solver
    * FVM formulation
    * Modeling the following major physical processes:

>       * Proppant-fluid slurry flow and mutlicomponent trasnport in fractures

Reviewer 5: ```suggestion
      * Proppant-fluid slurry flow and multicomponent transport in fractures
```
